DependencyInjection/CmfBlogExtension.php

```
@@ -35,53 +35,57 @@ public function load(array $configs, ContainerBuilder $container)
-        ), isset($config['class']) ? $config['class'] : array());
+        $this->loadPaginationIntegration($config, $container);
+    }
+
+    protected function loadPhpcrPersistence($config, XmlFileLoader $loader, ContainerBuilder $container)
+    {
+        $container->setParameter($this->getAlias().'.blog_basepath', $config['persistence']['phpcr']['blog_basepath']);
 
-        foreach ($config['class'] as $type => $classFqn) {
+        foreach ($config['persistence']['phpcr']['class'] as $type => $classFqn) {
             $container->setParameter(
-                $param = sprintf('cmf_blog.%s_class', $type),
+                $param = sprintf('cmf_blog.phpcr.%s.class', $type),
                 $classFqn
             );
         }
+
+        $loader->load('initializer-phpcr.xml');
+        $loader->load('doctrine-phpcr.xml');
     }
 
-    private function loadSonataAdmin($config, XmlFileLoader $loader, ContainerBuilder $container)
+    protected function loadSonataAdmin(array $config, XmlFileLoader $loader, ContainerBuilder $container)
     {
         $bundles = $container->getParameter('kernel.bundles');
-        if ('auto' === $config['use_sonata_admin'] && !isset($bundles['SonataDoctrinePHPCRAdminBundle'])) {
+        if (!isset($bundles['SonataDoctrinePHPCRAdminBundle'])) {
             return;
         }
 
-        $loader->load('blog-admin.xml');
-        $container->setParameter($this->getAlias() . '.blog_basepath', $config['blog_basepath']);
+        $loader->load('admin.xml');
     }
 
-    private function loadMenuIntegration($config, XmlFileLoader $loader, ContainerBuilder $container)
+    protected function loadMenuIntegration(array $config, XmlFileLoader $loader, ContainerBuilder $container)
     {
         $bundles = $container->getParameter('kernel.bundles');
-        if ('auto' === $config['integrate_menu']['enabled'] && !isset($bundles['CmfMenuBundle'])) {
+        if (!isset($bundles['CmfMenuBundle'])) {
             return;
```

Menu integration is explicit .. so we should throw an Exception here if the Bundle is not registered